driver-core/src/main/com/mongodb/internal/connection/DefaultClusterFactory.java

```diff
@@ -68,6 +73,18 @@ public Cluster createCluster(final ClusterSettings clusterSettings, final Server
 
         ClusterId clusterId = new ClusterId();
 
+        AsynchronousClusterEventListener clusterEventListener =
+                new AsynchronousClusterEventListener(clusterId, getClusterListener(originalClusterSettings),
+                        getServerListener(originalServerSettings), getServerMonitorListener(originalServerSettings));
+
+        ClusterSettings clusterSettings = ClusterSettings.builder(originalClusterSettings)
+                .clusterListenerList(singletonList(clusterEventListener))
```

Comment:
These new methods are already proving valuable!

Reply:
Nice! Pinging @ramasai1 so that he sees the fruits of labor :)